lib/component.js

```diff
@@ -616,12 +623,21 @@ class Component extends WebComponent {
       const attrType = attrSchema.type;
       let attrValue = null;
 
+      if (attrSchema.deprecatedMsg) {
+        console.warn(`${this}: attr '${attr}' is deprecated. ${attrSchema.deprecatedMsg}`);
+      }
+
       if (!this.hasAttribute(attr)) {
         if (attrType === `boolean` && attrSchema.default) {
           console.warn(
             `${this}: attr '${attr}' is defaulted to true. Default-true boolean attributes are deprecated and will be removed in a future release`,
           );
         }
+        if (attrSchema.required && attrType !== `boolean`) {
```

Comment:
if we throw real error, we don't need this.

Reply:
We still need the `if(attrSchema.required)` check, but we can leave off the addition boolean check.